File: scala/src/main/scala/Messages.scala
trait BankMessage {
  def accountId: String
}

case class Open(accountId: String) extends BankMessage
case class Close(accountId: String) extends BankMessage
case class Deposit(accountId: String, amount: Double) extends BankMessage
case class Withdrawal(accountId: String, amount: Double) extends BankMessage
case class GetBalance(accountId: String) extends BankMessage
case class Transfer(accountId: String, toId: String, amount: Double) extends BankMessage

File: csharp/AkkaBank/Messages.cs
namespace AkkaBank {
    public class BankMessage {
        public string AccountId { get; set; }
    }

    public class Close : BankMessage {}

    public class Deposit : BankMessage {
        public double Amount { get; set; }
    }

    public class GetBalance : BankMessage {}

    public class Open : BankMessage {}

    public class Transfer : BankMessage {
        public string ToId { get; set; }
        public double Amount { get; set; }
    }

    public class Withdrawal : BankMessage {
        public double Amount { get; set; }
    }
}